specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-04-01-preview/apimauthorizationproviders.json

```diff
@@ -0,0 +1,805 @@
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders": {
+      "get": {
+        "tags": [
+          "AuthorizationProvider"
+        ],
+        "operationId": "AuthorizationProvider_ListByService",
+        "description": "Lists a collection of authorization providers defined within a service instance.",
+        "x-ms-examples": {
+          "ApiManagementListAuthorizationProviders": {
+            "$ref": "./examples/ApiManagementListAuthorizationProviders.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "|     Field     |     Usage     |     Supported operators     |     Supported functions     |</br>|-------------|-------------|-------------|-------------|</br>| name | filter | ge, le, eq, ne, gt, lt | substringof, contains, startswith, endswith |</br>| displayName | filter | ge, le, eq, ne, gt, lt | substringof, contains, startswith, endswith |</br>"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/TopQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SkipQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A Collection of the Authorization Provider entities for the specified API Management service instance.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/AuthorizationProviderCollection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "./definitions.json#/definitions/AuthorizationProviderContract"
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}": {
```

Comment:
>authorizationProviderId [](http://example.com/codeflow?start=152&length=23)

Just wondering, is it really an ID or a name?

Reply:
@LFZ96 can you comment?

Reply:
It's a name

Reply:
Want to update the parameter naming to match that statement then?

Reply:
@TimLovellSmith Yes, I will check with my team. I'm new to the project and the person that has worked on this document has left the team. I remember having the same question and him giving an explanation for it. I will report back.

Reply:
@TimLovellSmith My understanding is that it's the identifier of the authorization provider that coincides with the name that the user gives it. For example, `google-provider-123`. It's unique and is used to identify the auth provider rather than something like a GUID. These entities are also close 1-to-1s of Connector team connectors/connections where the names of those are similarly `ConnectorId` and `ConnectionId`.